### DISCUSSION TITLE
Are there existing conventions for representing vectors and matrices?

### DISCUSSION THREAD

**Discussion by ZedThree** [pydata/xarray#7583] [Q&A]
I've not found a good reference for how people represent vectors and matrices in `xarray`. Take this example:

```py
import xarray as xr
import numpy as np

x = np.arange(4)
y = np.arange(5)
array = np.random.random((10, 4, 5))
vector = np.random.random((10, 4, 5, 2))
matrix = np.random.random((10, 4, 5, 2, 2))
time = np.arange(10)

ds = xr.Dataset(
  {
    "array": (["time", "x", "y"], array),
    "vector": (["time", "x", "y", "e1"], vector),
    "matrix": (["time", "x", "y", "e1", "e2"], matrix)
  }, coords={
    "time": time,
    "x": x,
    "y": y,
    "e1": ["x", "y"],
    "e2": ["x", "y"]
  }
)
```

Here `e1, e2` represent the set of spatial dimensions. Really, there is only one coordinate dimension here that we have to name twice for the matrix.

You could imagine splitting the vector out into some more complex type with named components `vector.x, vector.y`, but that feels like it wouldn't scale to matrices.

Is there a conventional way to represent a dimension that is itself some set of dimensions? Or to have multiple dimensions that refer to the same coordinate?

**Comment by dcherian** [comment]
See https://github.com/pydata/xarray/discussions/5775 I don't think there's any consensus.
